src/Qir/Execution/Tools/Executable/QirExecutable.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools.Driver;
+using Microsoft.Quantum.Qir.Utility;
+using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+namespace Microsoft.Quantum.Qir.Tools.Executable
+{
+    /// <summary>
+    /// Base for creating and running QIR-based executables.
+    /// </summary>
+    public abstract class QirExecutable : IQirExecutable
+    {
+        private const string DriverFileName = "driver";
+        private const string BytecodeFileName = "qir.bc";
+
+        public virtual string SourceDirectoryPath => "src";
+        public abstract string DriverFileExtension { get; }
+
+        protected FileInfo ExecutableFile { get; }
+
+        private readonly byte[] qirBytecode;
+        private readonly ILogger logger;
+        private readonly IQuantumExecutableRunner runner;
+        private readonly IQirDriverGenerator driverGenerator;
+        private readonly IQirExecutableGenerator executableGenerator;
+
+        internal QirExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger, IQirDriverGenerator driverGenerator, IQirExecutableGenerator executableGenerator, IQuantumExecutableRunner runner)
```

Comment:
Can you please add a `public` constructor that takes `executableFile`, `qirBytecode`, `driverGenerator` and an optional `logger`?
```csharp
public QirExecutable(FileInfo executableFile, byte[] qirBytecode, IQirDriverGenerator driverGenerator, ILogger? logger = null)
{
}
```

Reply:
It seems like this was not added in the latest iteration.